admin/base/views.py

```diff
@@ -28,3 +34,44 @@ def password_reset_confirm_custom(request, **kwargs):
             user.confirmed = True
             user.save()
     return response
+
+
+class GuidFormView(FormView):
+    form_class = GuidForm
+    template_name = None
+    object_type = None
+
+    def get_guid_object(self):
+        raise NotImplementedError
+
+    def __init__(self):
+        self.guid = None
+        super(GuidFormView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(GuidFormView, self).get(request, args, kwargs)
+        except AttributeError:
+            return page_not_found(self.request)
+
+    def get_context_data(self, **kwargs):
+        self.guid = self.request.GET.get('guid', None)
+        if self.guid is not None:
+            try:
+                guid_object = self.get_guid_object()
+            except AttributeError:
+                raise
+        else:
+            guid_object = None
```

Comment:
Just a style thing, I would set `guid_object = None` at the top of the method and lose the else. Then it's None by default and it'll not be None if get_guid_object works.